### PR TITLE
Added support for Azure v2 resource.

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Added support for Azure v2 np resource.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-13687

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+  aks_region                             = "region"
+  controller_cluster_id                  = "controllerId"
+  aks_cluster_name                       = "nodePoolName"
+  aks_infrastructure_resource_group_name = "aksInfrastructureResourceGroupName"
+  aks_resource_group_name                = "resourceGroup"
+  autoscale_is_enabled      = true
+  max_vcpu       = 121
+  max_memory_gib = 120
+  max_scale_down_percentage = 10
+  is_enabled = true
```
Comment (anuragsharma-123) [inline examples/basic/main.tf]: Confusing parameter. Not clear what it is for. Same for percentage.
is_automatic_autoscaler_headroom_enabled
automatic_autoscaler_headroom_percentage

File: examples/basic/main.tf [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+module "ocean-aks-np" {
   source = "../.."
+  # Credentials.
+  spotinst_token   = var.spotinst_token
+  spotinst_account = var.spotinst_account
+  # Configuration
+  name                                   = "clusterName"
+  aks_region                             = "region"
+  controller_cluster_id                  = "controllerId"
+  aks_cluster_name                       = "nodePoolName"
```
Comment (chandra1-n) [inline examples/basic/main.tf]: mention as aksClusterName instead of nodePoolName

File: examples/basic/main.tf [hunk 3]
```diff
@@ -1,8 +1,37 @@
...
+  aks_infrastructure_resource_group_name = "aksInfrastructureResourceGroupName"
+  aks_resource_group_name                = "resourceGroup"
+  autoscale_is_enabled      = true
+  max_vcpu       = 121
+  max_memory_gib = 120
+  max_scale_down_percentage = 10
```
Comment (anuragsharma-123) [inline examples/basic/main.tf]: prepend the field-name with autoscale_

File: examples/basic/main.tf [hunk 4]
```diff
@@ -1,8 +1,37 @@
...
+  spotinst_token   = var.spotinst_token
+  spotinst_account = var.spotinst_account
+  # Configuration
+  name                                   = "clusterName"
+  aks_region                             = "region"
+  controller_cluster_id                  = "controllerId"
+  aks_cluster_name                       = "nodePoolName"
+  aks_infrastructure_resource_group_name = "aksInfrastructureResourceGroupName"
+  aks_resource_group_name                = "resourceGroup"
+  autoscale_is_enabled      = true
```
Comment (anuragsharma-123) [inline examples/basic/main.tf]: is_autoscale_enabled

File: examples/basic/main.tf (hunk 1 above)
Comment (sharadkesarwani) [reply examples/basic/main.tf]: updated to autoscaler_headroom_automatic_is_enabled and autoscaler_headroom_automatic_percentage.

File: examples/basic/main.tf (hunk 2 above)
Comment (sharadkesarwani) [reply examples/basic/main.tf]: Updated to aksClusterName

File: examples/basic/main.tf (hunk 3 above)
Comment (sharadkesarwani) [reply examples/basic/main.tf]: updated to autoscaler_headroom_automatic_percentage.

File: examples/basic/main.tf (hunk 4 above)
Comment (sharadkesarwani) [reply examples/basic/main.tf]: updated to autoscaler_is_enabled .

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**